molgenis-data-mapper/src/main/java/org/molgenis/data/mapper/job/MappingJob.java

```diff
@@ -1,83 +1,64 @@
 package org.molgenis.data.mapper.job;
 
-import com.google.common.collect.Lists;
+import org.molgenis.data.DataService;
 import org.molgenis.data.jobs.Job;
 import org.molgenis.data.jobs.Progress;
+import org.molgenis.data.mapper.mapping.model.EntityMapping;
+import org.molgenis.data.mapper.mapping.model.MappingProject;
+import org.molgenis.data.mapper.mapping.model.MappingTarget;
+import org.molgenis.data.mapper.service.MappingService;
 import org.springframework.security.core.Authentication;
 import org.springframework.transaction.support.TransactionTemplate;
 
-import java.util.List;
-
-import static java.util.Objects.requireNonNull;
+import static org.molgenis.data.mapper.service.impl.MappingServiceImpl.MAPPING_BATCH_SIZE;
 
 public class MappingJob extends Job<Void>
 {
-	private final String username;
-	List<String> successfulAnnotators = Lists.newArrayList();
-	List<String> failedAnnotators = Lists.newArrayList();
-	Exception firstException = null;
+	private final MappingService mappingService;
+	private final MappingJobExecution mappingJobExecution;
+	private final DataService dataService;
 
-	public MappingJob(String username, Progress progress, Authentication userAuthentication,
-			TransactionTemplate transactionTemplate)
+	MappingJob(MappingJobExecution mappingJobExecution, Progress progress, Authentication userAuthentication,
```

Comment:
Probably this one is easier to unit test if you write out the separate arguments.